invokeai/backend/model_manager/install.py

```diff
@@ -0,0 +1,758 @@
+        if info.model_type == ModelType.Main:
+            if info.variant_type:
+                registration_data.update(variant=info.variant_type)
+            if info.format == ModelFormat.Checkpoint:
+                try:
+                    config_file = self._legacy_configs[info.base_type][info.variant_type]
+                    if isinstance(config_file, dict):  # need another tier for sd-2.x models
+                        if prediction_type := info.prediction_type:
+                            config_file = config_file[prediction_type]
+                        else:
+                            self._logger.warning(
+                                f"Could not infer prediction type for {model_path.stem}. Guessing 'v_prediction' for a SD-2 768 pixel model"
+                            )
+                            config_file = config_file[SchedulerPredictionType.VPrediction]
+                    registration_data.update(
+                        config=Path(self._app_config.legacy_conf_dir, str(config_file)).as_posix(),
+                    )
+                except KeyError as exc:
+                    raise InvalidModelException(
+                        "Configuration file for this checkpoint could not be determined"
+                    ) from exc
+        self._store.add_model(key, registration_data)
+        return key
+
+    def install_path(
+        self,
+        model_path: Union[Path, str],
+        overrides: Optional[Dict[str, Any]] = None,
+    ) -> str:  # noqa D102
+        model_path = Path(model_path)
+        info: ModelProbeInfo = self._probe_model(model_path, overrides)
+
+        dest_path = self._app_config.models_path / info.base_type.value / info.model_type.value / model_path.name
+        return self._register(
+            self._move_model(model_path, dest_path),
+            info,
+        )
+
+    def _move_model(self, old_path: Path, new_path: Path) -> Path:
+        if old_path == new_path:
+            return old_path
+
+        new_path.parent.mkdir(parents=True, exist_ok=True)
+
+        # if path already exists then we jigger the name to make it unique
+        counter: int = 1
+        while new_path.exists():
+            path = new_path.with_stem(new_path.stem + f"_{counter:02d}")
+            if not path.exists():
+                new_path = path
+            counter += 1
+        return old_path.replace(new_path)
+
+    def _probe_model(self, model_path: Union[Path, str], overrides: Optional[Dict[str, Any]] = None) -> ModelProbeInfo:
+        info: ModelProbeInfo = ModelProbe.probe(Path(model_path))
+        if overrides:  # used to override probe fields
+            for key, value in overrides.items():
+                try:
+                    setattr(info, key, value)  # skip validation errors
+                except Exception:
+                    pass
+        return info
+
+    def unregister(self, key: str):  # noqa D102
+        self._store.del_model(key)
```

Comment:
I noticed that this and a few other methods in this class are only used within the class. Are these intended to be private?

Reply:
This one is intended to be public; the ABC has a docstring and it is referred to in the code example at the top of the file. I'll check other public methods.